app/components/channel_actions/manage_members_label/index.tsx

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
+import withObservables from '@nozbe/with-observables';
+import {of as of$} from 'rxjs';
+import {combineLatestWith, switchMap} from 'rxjs/operators';
+
+import {General} from '@constants';
+import {observeChannel} from '@queries/servers/channel';
+import {queryUsersById} from '@queries/servers/user';
+
+import ManageMembersLabel from './manage_members_label';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+
+type OwnProps = WithDatabaseArgs & {
+    channelId: string;
+    userId: string;
+}
+
+const enhanced = withObservables(['channelId', 'userId'], ({channelId, userId, database}: OwnProps) => {
+    const users = queryUsersById(database, [userId]).fetch();
+
+    const channel = observeChannel(database, channelId);
+    const canRemove = channel.pipe(
+        combineLatestWith(users),
+        switchMap(([ch, u]) => {
+            const isDefaultChannel = ch?.name === General.DEFAULT_CHANNEL;
+            return of$(!isDefaultChannel || (isDefaultChannel && u[0].isGuest));
+        }),
```

Comment:
I wonder that instead of having to query the channel for each member in the list, if yo should pass the prop `isDefaultChannel` and have a query less per member

Reply:
great suggestion. This has been implemented and committed.